docs/examples/DataSet/Deeper dive on qcodes doNd functionalities with a focus on arrayed sweeping benefits.ipynb

```diff
@@ -0,0 +1,455 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Necessary imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qcodes import initialise_or_create_database_at\n",
+    "import time\n",
+    "import numpy as np\n",
+    "from qcodes.instrument.parameter import Parameter\n",
+    "from qcodes import load_or_create_experiment, Measurement,\\\n",
+    "load_by_run_spec\n",
+    "from qcodes.utils.dataset import doNd\n",
+    "from qcodes.dataset.plotting import plot_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating a database or loading it"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "initialise_or_create_database_at('test.db')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Point by point 2D sweep over x and y"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell is a naive way of generating an example 2D plot and time it. This uses single sweep points for both x and y directions, so it is quit slow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Starting experimental run with id: 40. \n",
+      "Meaurement took 287.8640215396881 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "xvals = np.linspace(-5, 5, 51)\n",
+    "yvals = np.linspace(-5, 5, 51)\n",
+    "xx, yy = np.meshgrid(xvals, yvals, indexing='ij')\n",
+    "zz = np.cos(xx) * np.sin(yy)\n",
+    "\n",
+    "def very_simple_2d_sweep():\n",
+    "    for i, x in enumerate(xvals):\n",
+    "        for j, y in enumerate(yvals):\n",
+    "            yield x, y, zz[i, j]\n",
+    "\n",
+    "# configure the qcodes setup\n",
+    "x = Parameter('x')\n",
+    "y = Parameter('y')\n",
+    "z = Parameter('z')\n",
+    "\n",
+    "exp = load_or_create_experiment('very_simple_2d_sweep', sample_name='no sample')\n",
+    "\n",
+    "# set up the measurement\n",
+    "meas = Measurement(exp)\n",
+    "meas.register_parameter(x)\n",
+    "meas.register_parameter(y)\n",
+    "meas.register_parameter(z, setpoints=(x, y))\n",
+    "meas.write_period = 2\n",
+    "\n",
+    "# and start recording\n",
+    "t1 = time.time()\n",
+    "with meas.run() as datasaver:\n",
+    "    for xval, yval, zval in very_simple_2d_sweep():\n",
+    "        datasaver.add_result(\n",
+    "            ('x', xval),\n",
+    "            ('y', yval),\n",
+    "            ('z', zval),\n",
+    "        )\n",
+    "        time.sleep(0.1)\n",
```

Comment:
i have a question about the "timings comparison". I clearly don't udnerstand something about this example notebook :)

so, if the goal is to "simulate" a measurement that take 0.1s per point vs a buffered measurement that takes 0.1s per buffer (array of y), then i'm not sure i get the point of this notebook because this comparison can be done without any code, right?

now, if the point is to demonstrate how to transition from point-per-point measurement to buffer-per-buffer measurement code using Measurement() and doNd(), then the notebook clearly lacks clarity on explaining its point. It should be very clear to the user what is being compared and what is demonstrated and how, otherwise i read this notebook as proving the obvious in a too abstract way.

Reply:
Actually, your question is because of that the initial goal of the notebook was to show the difference between Measurement and doNd. The point by point comparison with buffered added later, so that's might be the reason you can't find cohesion in the notebook. The main point is to show the use case of do1d and do2d compared to Measurement. So, timing and all buffered stuff were the last minute addition to have extra context in these comparisons.

That's said, do you think I should only focus on the Measurement and doNd or what?

Reply:
then it sounds like there should be two notebooks - one about Measurement vs doNd, and another one about buffered acquisitions (which would include a section showing that buffered acquisition with doNd functions also "works"). Would you agree?

Reply:
Sounds fine with me. I will change this notebook scope to only cover Measurement VS doNd. I will try to add sections to cover all doNd functions compared to Measurement. We are lacking do0d in this notebook.